src/solvers/flattening/boolbv.cpp

```diff
@@ -324,8 +334,8 @@ bvt boolbvt::convert_lambda(const exprt &expr)
   if(width==0)
     return conversion_failed(expr);
 
-  if(expr.operands().size()!=2)
-    throw "lambda takes two operands";
+  DATA_INVARIANT(
+    expr.operands().size() == 2, "lambda expression should have two operands");
```

Comment:
There currently exists a lambda expression type in a non-public repo. @peterschrammel: Should this be pulled out from there and added to `std_expr.h`?

Reply:
That's used in cegis. @pkesseli will remove this as soon as the cegis code has been cleaned up.

Reply:
I've raised an issue for that as well now: #3039